src/accordion/panel.js

```diff
@@ -90,62 +95,32 @@ class Panel extends React.Component<PanelPropsT, {isFocusVisible: boolean}> {
     } = this.props;
 
     const sharedProps = this.getSharedProps();
-    const {
-      PanelContainer: PanelContainerOverride,
-      Header: HeaderOverride,
-      Content: ContentOverride,
-      ToggleIcon: ToggleIconOverride,
-    } = overrides;
-
-    const isIconOverriden =
-      ToggleIconOverride &&
-      (typeof ToggleIconOverride === 'function' ||
-        typeof ToggleIconOverride.component === 'function');
-
-    const [PanelContainer, panelContainerProps] = getOverrides(
-      PanelContainerOverride,
-      StyledPanelContainer,
-    );
-    const [Header, headerProps] = getOverrides(HeaderOverride, StyledHeader);
-    const [Content, contentProps] = getOverrides(
-      ContentOverride,
-      StyledContent,
-    );
-    const [ToggleIcon, toggleIconProps] = getOverrides(
-      ToggleIconOverride,
-      StyledToggleIcon,
-    );
-
-    const toggleIconOverrides = mergeOverrides(
-      {Svg: ToggleIcon},
-      // $FlowFixMe
-      {Svg: ToggleIconOverride},
-    );
-
-    // it's a bit tricky ¯\_(ツ)_/¯
-    // we only want to use the theme overrides, if it was not override locally
-    const ToggleIconComponent = isIconOverriden
-      ? ToggleIcon
-      : expanded
-      ? CheckIndeterminateIcon
-      : PlusIcon;
+    const ToggleIconComponent = expanded ? CheckIndeterminateIcon : PlusIcon;
 
     return (
       <LocaleContext.Consumer>
         {locale => (
-          <PanelContainer {...sharedProps} {...panelContainerProps}>
+          <PanelContainer {...sharedProps} override={overrides.PanelContainer}>
             <Header
               tabIndex={0}
               role="button"
               aria-expanded={expanded}
               aria-disabled={disabled || null}
               {...sharedProps}
-              {...headerProps}
               {...(ariaControls ? {'aria-controls': ariaControls} : {})}
               onClick={this.onClick}
               onKeyDown={this.onKeyDown}
-              onFocus={forkFocus(headerProps, this.handleFocus)}
-              onBlur={forkBlur(headerProps, this.handleBlur)}
+              onFocus={
+                overrides.Header && overrides.Header.props
+                  ? forkFocus(overrides.Header.props, this.handleFocus)
+                  : this.handleFocus
+              }
+              onBlur={
```

Comment:
Is it guaranteed that this 'onBlur' handler won't get overwritten by `overrides.Header.props.onBlur`? Same for other event handlers.

Reply:
Good point, the blur fork will get clobbered if `onBlur` exists in override props. I will look into how to resolve. Other than these two handlers, we will want overrides to take precedent right? I had noticed that `headerProps` were provided before the event handlers on this component, but thought it was a bug.

Reply:
It looks like all the handlers/props that go after `{...headerProps}` in the old code are not supposed to be overwritten.

Reply:
I personally think we get into a very messy situation to selectively apply overrides. This change should be equivalent for `onClick` and `onKeyDown` since those prop overrides were never applied before. Now, that function props overrides are better supported, we can do something like below and get the best both situations.

```js
const overrides = {
  Header: {
    props: (props) => {
      return {
        ...props,
        onClick: (event) => {
          if (props.onClick) props.onClick(event);
          // add anything I want
        },
        onKeyDown: (event) => {
          if (props.onKeyDown) props.onKeyDown(event);
          // add anything I want
        },
        onFocus: (event) => {
          // maintains the fork
          if (props.onFocus) props.onFocus(event);
          // add anything I want
        },
        onBlur: (event) => {
          // maintains the fork
          if (props.onBlur) props.onBlur(event);
          // add anything I want
        },
      };
    },
  },
};
```

Reply:
I like it, but that would be a breaking change now. I think for the v10 tho this would be the way to move to. 
`onKeyDown` and `onClick` currently accepted as top-level props so they are applied, just no need to pass them through overrides.

Reply:
resolved and added a test: https://github.com/uber/baseweb/pull/3163/files#diff-71e348c99a67e7b6ca25cb442bd39f1dR111